modules/seqexec/server/src/main/scala/seqexec/server/gpi/GpiController.scala

```diff
@@ -246,10 +255,41 @@ object GpiController extends GpiLookupTables with GpiConfigEq {
   ): GpiController[F] =
     new AbstractGiapiInstrumentController[F, GpiConfig, GpiClient[F]](client)
     with GpiController[F] {
-      override val gdsClient: GdsClient[F] = gds
+      private implicit val eqResponse: Eq[Response] = Eq.fromUniversalEquals
+      override val gdsClient: GdsClient[F]          = gds
 
       override val name = "GPI"
 
+      private val GuidingChannel = GiapiStatus.GpiGuiding.statusItem
+
+      // Check if we are guiding reading from the local status db
+      private def guiding: F[GpiGuiding] =
+        client.statusDb
+          .value(GuidingChannel)
+          .map(StatusValue.intValue(_).map(x => GpiGuiding.fromInt(x.toInt)))
+          .ensure(new SeqexecFailure.Execution(s"Channel $GuidingChannel should be an int"))(
+            _.isDefined)
+          .map { _.orNull } // orNull lets us typecheck but it will never be used due to the `ensure` call above
+          .adaptError {
+            case GiapiException(message) => SeqexecFailure.Execution(message) }
+
+      override def alignAndCalib: F[Unit] =
+        guiding
+          .map(_ === GpiGuiding.NotGuiding)
```

Comment:
I'm having second thoughts on this check. We have had the philosophy of requiring instruments to ensure we do operations safely rather than on the seqexec. I verified the instrument code and indeed this is checked at the lower level
@jluhrs thoughts?